src/lib/serverCommon.ts

```diff
@@ -14,12 +27,16 @@ export function validateClass(schema: { definitions: any }, className: string) {
       throw new ValidationError('Bad Request', [{ message: 'Could not parse body', method }]);
     }
 
-    const { context, args } = body;
-
-    if (contextValidator && !contextValidator(context)) {
-      throw new ValidationError('Bad Request', contextValidator.errors);
+    const { context, args, ...rest } = body;
+    if (args && Object.keys(rest).length === 0) {
+      delete body.args;
+      Object.assign(body, args);
+      if (context) {
+        body.ctx = context;
+        delete body.context;
+      }
     }
-    if (!validator(args)) {
+    if (!validator(body)) {
```

Comment:
I think we need to come back to the additional properties discussion, since previous version allowed unlisted properties on the body (which were not possible to be used, and yet).

Reply:
Thinking about it some more, it does mean that the protocol becomes more difficult to extend (also the protocol is not versioned so we use heuristics now - and will be forced to use them in the future).

Reply:
Sure